src/lsptoolshost/dotnetRuntimeExtensionResolver.ts

```diff
@@ -0,0 +1,104 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { HostExecutableInformation } from "../shared/constants/HostExecutableInformation";
+import { IHostExecutableResolver } from "../shared/constants/IHostExecutableResolver";
+import { PlatformInformation } from "../shared/platform";
+import { Options } from "../shared/options";
+import { existsSync } from 'fs';
+import { CSharpExtensionId } from '../constants/CSharpExtensionId';
+
+export const DotNetRuntimeVersion = '7.0';
+
+interface IDotnetAcquireResult {
+    dotnetPath: string;
+}
+
+/**
+ * Resolves the dotnet runtime for a server executable from given options and the dotnet runtime VSCode extension.
+ */
+export class DotnetRuntimeExtensionResolver implements IHostExecutableResolver {
+    constructor(
+        private platformInfo: PlatformInformation,
+        /**
+         * This is a function instead of a string because the server path can change while the extension is active (when the option changes).
+         */
+        private getServerPath: (options: Options, platform: PlatformInformation) => string) { }
```

Comment:
A couple reasons for doing it this way
1.  As mentioned in the comment it needs to be a function for when the server path changes (from the option).  One `IHostExecutableResolver` is created during activation for roslyn, and re-use across server restarts.
2.  It didn't seem appropriate to update `getHostExecutableInfo` because not all implementations of `IHostExecutableResolver` require a server path (for example used in O# with dotnet and mono as well).  Its more of a property of this specific resolver.